frontend/taipy-gui/src/utils/formatConversion.ts

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2021-2024 Avaiga Private Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+const FORMAT_REPLACE_REGEX = /%([0-9]*)([.][0-9]+)?([bdieufgosxX])/g;
+
+/**
+ * Converts a sprintf format string to a D3 format string.
+ *
+ * This function takes a sprintf format string as input and returns a D3 format string.
+ * The conversion is done by replacing each sprintf format specifier with the corresponding D3 format specifier.
+ * If the input format string is undefined, the function returns undefined.
+ * If an error occurs during the conversion, the function logs the error to the console and returns undefined.
+ *
+ * @param format - The sprintf format string to convert.
+ * @returns The converted D3 format string, or undefined if the input is undefined or if an error occurs.
+ */
+const sprintfToD3Converter = (format: string) => {
+    try {
+        return format?.replace(FORMAT_REPLACE_REGEX, (match, width, precision, type) => {
+            switch (type) {
```

Comment:
This switch statement can be greatly reduced:
```
case "b":
case "d":
case "i":
   ...
   return type;
```

Reply:
resolved